orchagent/dash/dashhaorch.cpp

```diff
@@ -0,0 +1,491 @@
+    ha_set_attr_list[6].value.u32 = entry.dp_channel_probe_interval_ms();
+
+    ha_set_attr_list[7].id = SAI_HA_SET_ATTR_DP_CHANNEL_PROBE_FAIL_THRESHOLD;
+    ha_set_attr_list[7].value.u32 = entry.dp_channel_probe_fail_threshold();
+
+    status = sai_dash_ha_api->create_ha_set(&sai_ha_set_oid,
+                                                gSwitchId,
+                                                attr_count,
+                                                ha_set_attr_list);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to create HA Set object in SAI for %s", key.c_str());
+        task_process_status handle_status = handleSaiCreateStatus((sai_api_t) SAI_API_DASH_HA, status);
+        if (handle_status != task_success)
+        {
+            return parseHandleSaiStatusFailure(handle_status);
+        }
+    }
+    m_ha_set_entries[key] = HaSetEntry {sai_ha_set_oid, entry};
+    SWSS_LOG_NOTICE("Created HA Set object for %s", key.c_str());
+
+    return true;
+}
+
+bool DashHaOrch::removeHaSetEntry(const std::string &key)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = m_ha_set_entries.find(key);
+
+    if (it == m_ha_set_entries.end())
+    {
+        SWSS_LOG_WARN("HA Set entry does not exist for %s", key.c_str());
+        return true;
+    }
+
+    sai_status_t status = sai_dash_ha_api->remove_ha_set(it->second.ha_set_id);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to remove HA Set object in SAI for %s", key.c_str());
+        task_process_status handle_status = handleSaiRemoveStatus((sai_api_t) SAI_API_DASH_HA, status);
+        if (handle_status != task_success)
+        {
+            return parseHandleSaiStatusFailure(handle_status);
+        }
+    }
+    m_ha_set_entries.erase(it);
+    SWSS_LOG_NOTICE("Removed HA Set object for %s", key.c_str());
+
+    return true;
+}
+
+void DashHaOrch::doTaskHaSetTable(ConsumerBase &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple tuple = it->second;
+        const auto& key = kfvKey(tuple);
+        const auto& op = kfvOp(tuple);
+
+        if (op == SET_COMMAND)
+        {
+            dash::ha_set::HaSet entry;
+
+            if (!parsePbMessage(kfvFieldsValues(tuple), entry))
+            {
+                SWSS_LOG_WARN("Requires protobuf at HaSet :%s", key.c_str());
+                it = consumer.m_toSync.erase(it);
+                continue;
+            }
+
+            if (addHaSetEntry(key, entry))
+            {
+                it = consumer.m_toSync.erase(it);
+            }
+            else
+            {
+                it++;
+            }
+
+        }
+        else if (op == DEL_COMMAND)
+        {
+            if(removeHaSetEntry(key))
+            {
+                it = consumer.m_toSync.erase(it);
+            }
+            else
+            {
+                it++;
+            }
+        }
```

Comment:
Could you also add changes to write result of operations back to DPU_APPL_STATE_DB? Refer https://github.com/sonic-net/SONiC/pull/1759 and https://github.com/sonic-net/sonic-swss/pull/3490

Reply:
Addressed.